File: mq-demos/zmq_reply.py
# python3 zmq_reply.py 

import json
import zmq

context = zmq.Context()
socket = context.socket(zmq.REP)
socket.bind("tcp://*:5555")

while True:
    request = json.loads(socket.recv().decode('utf-8'))
    length = len(request['string'])
    response = {'length': length}
    socket.send(json.dumps(response).encode('utf-8'))

File: mq-demos/src/zmq_reply.rs
// cargo run --bin zmq_reply

#[macro_use]
extern crate json;

fn main() {
    let context = zmq::Context::new();
    let responder = context.socket(zmq::REP).unwrap();

    assert!(responder.bind("tcp://*:5555").is_ok());

    let mut msg = zmq::Message::new();
    loop {
        responder.recv(&mut msg, 0).unwrap();
        let request = json::parse(msg.as_str().unwrap()).unwrap();
        let length = request["string"].as_str().unwrap().len();
        let response = object!{
            "length": length
        };
        responder.send(json::stringify(response).as_bytes(), 0).unwrap();
    }
}
